src/trunk.c

```diff
@@ -4616,6 +4616,19 @@ trunk_btree_pack_req_init(trunk_handle   *spl,
                        spl->heap_id);
 }
 
+static void
+trunk_compact_bundle_cleanup_iterators(trunk_handle           *spl,
+                                       merge_iterator        **merge_itor,
+                                       uint64                  num_branches,
+                                       trunk_btree_skiperator *skip_itor_arr)
+{
+   platform_status rc = merge_iterator_destroy(spl->heap_id, merge_itor);
+   platform_assert_status_ok(rc);
+   for (uint64 i = 0; i < num_branches; i++) {
+      trunk_btree_skiperator_deinit(spl, &skip_itor_arr[i]);
+   }
+}
```

Comment:
I like / prefer to see such `deinit` functions NULL'ing out their input handles, so that we don't have stale ptrs in the caller upon returning.

How would you feel about NULL'ing out input `skip_itor_arr` before exiting?

Reply:
Done (in debug mode)